dpks/quant_matrix.py

```diff
@@ -459,121 +461,205 @@ def compare_groups(
+        if scale:
+            X = scaler.transform(X)
 
-                selector.rank_features(X, Y, classifier)
+        classifier = Classifier(classifier=classifier, shap_algorithm=shap_algorithm)
 
-                feature_rank_values = selector.ranking_.tolist()
-                for index in drop_indexes:
-                    feature_rank_values.insert(index, np.nan)
-                self.quantitative_data.obs["FeatureRank"] = feature_rank_values
+        classifier.interpret(X)
 
-                self.selector = selector
+        shap_values = classifier.feature_importances_.tolist()
+
+        self.quantitative_data.obs["SHAP"] = shap_values
+
+        self.shap = classifier.shap_values
+        self.transformed_data = X
 
         return self
 
+    def train(
+        self,
+        classifier,
+        scaler: Any = None,
+        shap_algorithm: str = "auto",
+        scale: bool = True,
+        validate: bool = True,
+        scoring: str = "accuracy",
+    ) -> TrainResult:
+
+        X = format_data(self)
+        y = encode_labels(self.quantitative_data.var["group"].values)
+
+        if scale:
+            if scaler:
+                X = scaler.fit_transform(X)
+            else:
+                scaler = StandardScaler()
+                X = scaler.fit_transform(X)
+
+        classifier = Classifier(classifier=classifier, shap_algorithm=shap_algorithm)
+
+        validation_result = np.array([])
+
+        if validate:
+            validation_result = cross_val_score(classifier, X, y, scoring=scoring)
+
+        classifier.fit(X, y)
+
+        return TrainResult(classifier, scaler, validation_result)
+
+    def optimize(
+        self,
+        classifier,
+        param_search_method: str,
+        param_grid: dict,
+        shap_algorithm: str = "auto",
+        scale: bool = True,
+        threads: int = 1,
+        random_state: int = 42,
+        folds: int = 3,
+        verbose: Union[bool, int] = False,
+        **kwargs: Union[dict, int, str, bool],
+    ) -> ParamSearchResult:
+
+        X = format_data(self)
+        y = encode_labels(self.quantitative_data.var["group"].values)
+
+        scaler = None
+
+        if scale:
+            scaler = StandardScaler()
+            X = scaler.fit_transform(X)
+
+        result = None
+
+        if param_search_method == "genetic":
+
+            gas = GeneticAlgorithmSearch(
+                classifier,
+                param_grid=param_grid,
+                threads=threads,
+                folds=folds,
+                n_survive=kwargs.get("n_survive", 5),
+                pop_size=kwargs.get("pop_size", 10),
+                n_generations=kwargs.get("n_generations", 20),
+                verbose=verbose,
+            )
+            parameter_populations = gas.fit(X, y)
+
+            result = ParamSearchResult(
+                classifier=gas.best_estimator_,
+                result=parameter_populations,
+            )
+
+        elif param_search_method == "grid":
```

Comment:
This is not grid search, this is randomized search. Change keyword to randomize. We should also add gridsearch